mysql/azuremysql/azuremysql.go

```diff
@@ -48,6 +61,45 @@ type Params struct {
 	TraceOpts []ocsql.TraceOption
 }
 
+// URLOpener opens Azure MySQL URLs
+// like "azuremysql://user:password@mydemoserver.mysql.database.azure.com:3306/mydb".
+type URLOpener struct {
+	// CertSource specifies how the opener will obtain authentication information.
+	// CertSource must not be nil.
+	CertSource CertPoolProvider
+	// TraceOpts contains options for OpenCensus.
+	TraceOpts []ocsql.TraceOption
+}
+
+// Scheme is the URL scheme azuremysql registers its URLOpener under on
+// mysql.DefaultMux.
+const Scheme = "azuremysql"
+
+func init() {
+	gcmysql.DefaultURLMux().RegisterMySQL(Scheme, &URLOpener{})
+}
+
+// OpenMySQLURL opens a new Azure database connection wrapped with OpenCensus instrumentation.
+func (uo *URLOpener) OpenMySQLURL(ctx context.Context, u *url.URL) (*sql.DB, error) {
+	if uo.CertSource == nil {
+		return nil, fmt.Errorf("azuremysql: URLOpener CertSource is nil")
+	}
+	password, _ := u.User.Password()
+	splitedHost := strings.Split(u.Host, ".")
+	params := &Params{
+		// Take only the server name from host
+		ServerName: splitedHost[0],
+		User:       u.User.Username(),
+		Password:   password,
+		// Remove "/" from DBPath
+		Database:  strings.Replace(u.Path, "/", "", -1),
```

Comment:
This needs the same `strings.TrimPrefix(u.Path, "/")` logic as in the other places.